api/topology/src/main/java/quarks/topology/TWindow.java

```diff
@@ -105,8 +129,85 @@ Licensed to the Apache Software Foundation (ASF) under one
+     * insertion order from oldest to newest.  The list is stable
+     * during the aggregator invocation.  
+     * The list will be empty if the partition is empty.
+     * </P>
+     * <P> 
+     * A non-null {@code aggregator} result is added to the returned stream.
+     * </P>
+     * <P>
+     * Thus the returned stream will contain a sequence of tuples where the
+     * most recent tuple represents the most up to date aggregation of a
+     * partition.
+     *
+     * @param <U> Tuple type
+     * @param period how often to invoke the aggregator
+     * @param unit TimeUnit for {@code period}
+     * @param aggregator
+     *            Logic to aggregation a partition.
+     * @return A stream that contains the latest aggregations of partitions in this window.
+     * 
+     * @see #aggregate(BiFunction)
+     */
+    <U> TStream<U> timedAggregate(long period, TimeUnit unit, BiFunction<List<T>, K, U> aggregator);
     
     /**
+     * Declares a stream that represents a 
+     * timer triggered batched aggregation of
+     * partitions in this window. 
+     * <P>
+     * Periodically trigger an invocation of
+     * {@code batcher.apply(tuples, key)}, where {@code tuples} is
+     * a {@code List<T>} containing all the tuples in the partition in
+     * insertion order from oldest to newest  The list is stable
+     * during the batcher invocation.
+     * The list will be empty if the partition is empty.
+     * <P>
+     * A non-null {@code batcher} result is added to the returned stream.
+     * The partition's contents are cleared after a batch is processed.
+     * </P>
+     * <P>
+     * Thus the returned stream will contain a sequence of tuples where the
+     * most recent tuple represents the most up to date aggregation of a
+     * partition.
+     * 
+     * @param <U> Tuple type
+     * @param period how often to invoke the batcher
+     * @param unit TimeUnit for {@code period}
+     * @param batcher
+     *            Logic to aggregation a partition.
+     * @return A stream that contains the latest aggregations of partitions in this window.
+     * 
+     * @see #batch(BiFunction)
+     */
+    <U> TStream<U> timedBatch(long period, TimeUnit unit, BiFunction<List<T>, K, U> batcher);
+    
+    /**
+     * Declares a stream that represents an aggregation of
+     * partitions in this window using the specified {@link Window}.
+     * <P>
+     * This method makes it easier to create aggregation streams
+     * using the lower level {@code Window} API to construct and
+     * supply windows with configurations not directly exposed by {@code TWindow}.
+     * 
+     * @param window the window implementation
+     * @param aggregator the aggregation function
+     *  
+     * @return A stream that contains the latest aggregations of partitions in this window.
+     */
+    <U, L extends List<T>> TStream<U> process(Window<T,K,L> window, BiFunction<List<T>, K, U> aggregator);
```

Comment:
Can this be explained a bit more, how does the passed in window relate to the window this is defined on?